### PR TITLE
Jupyter notebook

### PR DESCRIPTION
**Pull request by zoybai**
Add a simple parser and jupyter plotting script:

The default plotting script utilizes jupyter-notebook, matplotlib, seaborn
and pandas under python3 environment. For Ubuntu on x86_64 machine, the
following packages have to be installed:
apt install build-essential python3 python3-pip jupyter-notebook

For aarch64 machine, additional packages are also needed:
apt install pkg-config libfreetype6-dev python3-scipy

Then pip3 can install all plotting related libraries with the following cmd:
pip3 install matplotlib seaborn pandas numpy

Note, seaborn has to be installed without scipy as dependency on aarch64:
pip3 install seaborn --no-dependencies

The jupyter-notebook can be started with:
jupyter-notebook --ip 0.0.0.0 --port=8888

Now any browser should be able to access the jupyter notebook called:
lockhammer-jupyter-notebook.ipynb

Start a browser, with IP address set to the jupyter server IP and port 8888:
e.g. http://example.test.com:8888

Click the notebook named lockhammer-jupyter-notebook.ipynb, run each cell one
by one and jupyter should be able to generate the png graph locally.

### CODE REVIEW

**Comment by geoffreyblake** [comment]
Can one of the admins verify this patch?

File: benchmarks/lockhammer/scripts/lockhammer-jupyter-notebook.ipynb [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    "        \n",
+    "        # because lmplot doesn't plot lines, we have to use plot_lines_only to plot them\n",
+    "        sns.lmplot(x=\"num_threads\", y=param_name, hue=\"host\", data=tidy_df, x_estimator=np.median, x_ci=50,\n",
+    "                          height=10, aspect=2, fit_reg=False, markers=default_markers[:len(lock_hosts)], scatter_kws={\"s\": default_marker_size})\n",
+    "        \n",
+    "        plt.title(\"lockhammer workload: {}, critical_time: {}, parallel_time: {}\".format(workload, param[1], param[3]))\n",
+    "        plt.xlim(0, exectx_count_max)\n",
+    "        plt.xticks(np.arange(0, exectx_count_max+1, exectx_count_gap))\n",
+    "        \n",
+    "        if workload == \"swap_mutex\":\n",
```
Comment (geoffreyblake) [inline benchmarks/lockhammer/scripts/lockhammer-jupyter-notebook.ipynb]: Is this necessary for an example plotting script?

File: benchmarks/lockhammer/scripts/lockhammer-jupyter-notebook.ipynb [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "# use lmplot (not catplot) to plot raw system latencies\n",
+    "# TODO: pandas is single-threaded and very slow, plan to use dask in the future\n",
```
Comment (geoffreyblake) [inline benchmarks/lockhammer/scripts/lockhammer-jupyter-notebook.ipynb]: Should remove this TODO.

**Comment by zoybai** [comment]
Newer patch rebased to integration and merged with newer README.rst

**Comment by zoybai** [comment]
The new jupyter notebook script has been verified on an aarch64 machine.

**Pull request merged**